SharpSource/SharpSource/Diagnostics/UnnecessaryEnumerableMaterializationAnalyzer.cs

```diff
@@ -13,9 +13,7 @@ namespace SharpSource.Diagnostics;
 [DiagnosticAnalyzer(LanguageNames.CSharp)]
 public class UnnecessaryEnumerableMaterializationAnalyzer : DiagnosticAnalyzer
 {
-    private static readonly HashSet<string> MaterializingOperations = new(){
-        "ToList", "ToArray", "ToHashSet"
-    };
+    private static readonly ImmutableArray<string> MaterializingOperations = ImmutableArray.Create("ToList", "ToArray", "ToHashSet");
```

Comment:
Might be useful to add a comment here to explain an `ImmutableArray` was used because of the small number of values